src/relay/api/app.py

```diff
@@ -80,117 +93,128 @@ def ApiApp(trustlines):
     def add_resource(resource, url):
         api.add_resource(resource, url, resource_class_args=[trustlines])
 
+    # Always enabled
     api.add_resource(Version, "/version")
-    add_resource(NetworkList, "/networks")
-    add_resource(Network, "/networks/<address:network_address>")
-    add_resource(UserList, "/networks/<address:network_address>/users")
-    add_resource(EventsNetwork, "/networks/<address:network_address>/events")
-    add_resource(
-        UserAccruedInterestList,
-        "/networks/<address:network_address>/users/<address:user_address>/interests",
-    )
-    add_resource(
-        TrustlineAccruedInterestList,
-        "/networks/<address:network_address>/users/<address:user_address>/interests/<address:counterparty_address>",
-    )
-    add_resource(
-        User, "/networks/<address:network_address>/users/<address:user_address>"
-    )
-    add_resource(
-        ContactList,
-        "/networks/<address:network_address>/users/<address:user_address>/contacts",
-    )
-    add_resource(
-        TrustlineList,
-        "/networks/<address:network_address>/users/<address:user_address>/trustlines",
-    )
-    add_resource(
-        Trustline,
-        "/networks/<address:network_address>/users/<address:a_address>/trustlines/<address:b_address>",
-    )
-    add_resource(
-        MaxCapacityPath, "/networks/<address:network_address>/max-capacity-path-info"
-    )
-    add_resource(
-        UserEventsNetwork,
-        "/networks/<address:network_address>/users/<address:user_address>/events",
-    )
-    add_resource(Path, "/networks/<address:network_address>/path-info")
-    add_resource(
-        CloseTrustline, "/networks/<address:network_address>/close-trustline-path-info"
-    )
-    add_resource(UserEvents, "/users/<address:user_address>/events")
-    add_resource(TransactionInfos, "/users/<address:user_address>/txinfos")
-    add_resource(Balance, "/users/<address:user_address>/balance")
-    add_resource(UserTrustlines, "/users/<address:user_address>/trustlines")
-
     add_resource(Block, "/blocknumber")
-    add_resource(Relay, "/relay")
 
-    if trustlines.enable_relay_meta_transaction:
+    if ApiType.STATUS in enabled_apis:
```

Comment:
**This is only an idea that comes into my mind. It must not be accepted or discussed. Evaluate it yourself and feel free to close the conversation.**

I would argue that code could be more clean here. The function is extraordinary long and hard to read. The chance is likely that you miss something while jumping over large chunks. This part of the function has for me the higher level idea: for all enabled API types add the related resources to the api. It should not be the responsibility of this function to do the mapping. So my suggestion would be to reduce that to something that exactly put that idea into code:

```py
for api_type in enabled_apis:
    add_resources_for_type(api, api_type)
```

Resources are added by the tuple of their implementation and a path string. This could be moved into a mapping of the `ApiType` enum to a list of such tuples. Thereby this would not simply move the verbose code from above. The `add_resources_for_type` function could be as simple as:

```py
def add_resources_for_type(api: Api, api_type: ApiType) -> None:
    for resource in api_type.get_associated_resources():
        api.add_resource(resource.implementation, resource.path)
```

Reply:
Could be a bit nicer, but is nitpicking I would say. The function is setting up the api on a high level by turning on certain api endpoints, I think this is fine. Also the nesting is just one blevel deep.

Reply:
If we really need this, we can easily do this later.

Reply:
:+1: